src/zero_copy/primitive.rs

```diff
@@ -1,7 +1,57 @@
+//! Parser primitives that accept specific token patterns.
+//!
+//! *“These creatures you call mice, you see, they are not quite as they appear. They are merely the protrusion into
+//! our dimension of vastly hyperintelligent pandimensional beings.”*
+//!
+//! Chumsky parsers are created by combining together smaller parsers. Right at the bottom of the pile are the parser
+//! primitives, a parser developer's bread & butter. Each of these primitives are very easy to understand in isolation,
+//! usually only doing one thing.
+//!
+//! ## The Important Ones
+//!
+//! - [`just`]: parses a specific input or sequence of inputs
+//! - [`filter`]: parses a single input, if the given filter function returns `true`
+//! - [`end`]: parses the end of input (i.e: if there any more inputs, this parse fails)
+
 use super::*;
 
+/// See [`end`].
 pub struct End<I: ?Sized>(PhantomData<I>);
 
+/// A parser that accepts only the end of input.
+///
+/// This parser is very useful when you wish to force a parser to consume *all* of the input. It is typically combined
+/// with [`Parser::then_ignore`].
+///
+/// The output type of this parser is `()`.
+///
+/// # Examples
+///
+/// ```
+/// # use chumsky::prelude::*;
```

Comment:
hmm shouldn't you be using the zero copy prelude instead?
```suggestion
/// # use chumsky::zero_copy::prelude::*;
```
(same for ALL examples)

Reply:
Correct, I haven't yet updated most of the examples. I'm currently working on documenting things new in zero_copy, then once that's done, I plan to do a massive search-and-replace before cleaning up all the errors that causes